src/test/scala/com/crealytics/spark/excel/EncryptedReadSuite.scala

```diff
@@ -27,36 +27,44 @@ class EncryptedReadSuite extends FunSpec with DataFrameSuiteBase with Matchers {
 
   lazy val expected = spark.createDataFrame(expectedData, simpleSchema)
 
-  def readFromResources(path: String, password: String, maxRowsInMemory: Option[Int] = None): DataFrame = {
+  def readFromResources(
+    path: String,
+    password: String,
+    dataAddress: String,
+    maxRowsInMemory: Option[Int] = None
+  ): DataFrame = {
     val url = getClass.getResource(path)
     val reader = spark.read
-      .excel(
-        dataAddress = s"Sheet1!A1",
-        treatEmptyValuesAsNulls = true,
-        workbookPassword = password,
-        inferSchema = true
-      )
-    val withMaxRows = maxRowsInMemory.fold(reader)(rows => reader.option("maxRowsInMemory", s"$rows"))
+      .excel(dataAddress = dataAddress, treatEmptyValuesAsNulls = true, workbookPassword = password, inferSchema = true)
+    val withMaxRows = maxRowsInMemory.fold(reader)(rows => reader.option("maxRowsInMemory", rows))
     withMaxRows.load(url.getPath)
   }
 
   describe("spark-excel") {
     it("should read encrypted xslx file") {
-      val df = readFromResources("/spreadsheets/simple_encrypted.xlsx", "fooba")
+      val df = readFromResources("/spreadsheets/simple_encrypted.xlsx", "fooba", "Sheet1!A1")
 
       assertDataFrameEquals(expected, df)
     }
 
     it("should read encrypted xlsx file with maxRowsInMem=10") {
-      val df = readFromResources("/spreadsheets/simple_encrypted.xlsx", "fooba", maxRowsInMemory = Some(10))
+      val df =
+        readFromResources("/spreadsheets/simple_encrypted.xlsx", "fooba", "Sheet1!A1", maxRowsInMemory = Some(10))
 
       assertDataFrameEquals(expected, df)
     }
 
     it("should read encrypted xls file") {
-      val df = readFromResources("/spreadsheets/simple_encrypted.xls", "fooba")
+      val df = readFromResources("/spreadsheets/simple_encrypted.xls", "fooba", "Sheet1!A1")
 
       assertDataFrameEquals(expected, df)
     }
+
+    it("should read all xls file") {
+      val df = readFromResources("/spreadsheets/simple_encrypted.xls", "fooba", "Sheet1[#All]")
```

Comment:
Does this really work?
`Sheet1[#All]` means "take all columns of **table** `Sheet1`, but in the `simple_encrypted.xls` I don't see a table called `Sheet1`

Reply:
I didn't work before ("HSSFWorkbook cannot be cast to XSSFWorkbook" exception was thrown). But after this change it [does](https://github.com/crealytics/spark-excel/pull/140/commits/0aa46b3b48d598e18975b204e23ba065a95d28c5#diff-ed61aecb320eeee3fbc83ffe9d83f57eR159).
I agree, this is not what is described in the docs, but having ClassCastException is not very clear. Tbh, I didn't know difference between table and sheet when I started fixing ClassCastException.
I can revert this part if needed.

Reply:
Dear @nightscape and @EnverOsmanov,

I'm facing same exception when loading xls format. I'm not sure this fix is merged to master or not?

Reply:
Try `0.13.3+9-bea39dbd-SNAPSHOT`. 
There is some issue with releasing new versions using Github, that's why the latest is SNAPSHOT.

Reply:
Thanks @EnverOsmanov . But are there any issues if I using another data address format "'My Sheet'!A1"? Is it the same with "My_Table[#All]" ?

Reply:
I reviewed again, this PR wasn't merged. In my work cases, I use "sheet and cell" format ("Data!A3") and cell-only format ("B4").

Reply:
Yes, I did the same and everything worked fine till now. Thanks @EnverOsmanov for great support very much!